### PR TITLE
#849 Fixes to ClusterRouterGroup

### PR DESCRIPTION
**Pull request by annymsMthd**

### CODE REVIEW

File: src/core/Akka/Routing/RoutedActorCell.cs [hunk 1]
```diff
@@ -126,20 +122,17 @@ public override void Start()
...
+                        AddRoutees(Vector.Fill<Routee>(nrOfRoutees)(() => pool.NewRoutee(_routeeProps, this)));
                 })
-                .With<Group>(r =>
+                .With<Group>(group =>
                 {
-                    var routees = _routerConfig.GetRoutees(this).ToArray();
-                    AddRoutees(routees);
+                    var paths = group.Paths;
+                    if (paths.NonEmpty())
+                        AddRoutees(paths.Select(p => group.RouteeFor(p, this)).ToList());
```
Comment (annymsMthd) [inline src/core/Akka/Routing/RoutedActorCell.cs]: Down the rabbit hole.

**Comment by annymsMthd** [comment]
Going to squash these commits when i get home tonight. @Aaronontheweb mind giving this a look over when you can?

**Comment by Aaronontheweb** [comment]
yeessir - can do! I will personally be thrilled to see this spec pass. Been one that's eluded me for a while!

**Comment by annymsMthd** [comment]
I branched off this and merged in my app domain changes. Makes it really
easy with a debugger attached

**Comment by Aaronontheweb** [comment]
once we get an overload with the `Scheduler` running on a `DedicatedThreadPool` things should be easier for the `AppDomain` implementation - I strongly suspect threadpool saturation was at play with the inconsistent behavior I got.

**Comment by annymsMthd** [comment]
I'm going to be opening up a PR for it soon just to keep it around until we can get them to consistently pass. I also added the teamcity output.

File: src/core/Akka/Util/Internal/ArrayExtensions.cs [hunk 2]
```diff
@@ -27,6 +27,16 @@ public static bool IsNullOrEmpty(this Array obj)
         }
 
         /// <summary>
+        /// Determines if an array is not null or empty.
+        /// </summary>
+        /// <param name="obj">The array to check.</param>
+        /// <returns>True if not null or empty, false otherwise.</returns>
+        public static bool NonEmpty(this Array obj)
```
Comment (Aaronontheweb) [inline src/core/Akka/Util/Internal/ArrayExtensions.cs]: Wouldn't linq `.Any()` work for this? I just realized that right after I closed the PR :\

File: src/core/Akka/Util/Vector.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Akka.Util
+{
+    public static class Vector
```
Comment (Aaronontheweb) [inline src/core/Akka/Util/Vector.cs]: Could you add some XML-DOC comments for this on a subsequent PR @annymsMthd ?

Comment (annymsMthd) [inline src/core/Akka/Util/Vector.cs]: you got it

**Pull request merged**